### PR TITLE
feat: test whether page config works well before taking snapshots

### LINKED ISSUE #107: Test whether page config works well before taking snapshots

### PR DESCRIPTION
**Pull request by lizimeow**
image
image
image

Ping
image
image

close #107 

this PR includes #8

### CODE REVIEW

File: packages/platform-server/src/modules/snapshot/service.ts [hunk 1]
```diff
@@ -307,6 +356,31 @@ export class SnapshotService implements OnApplicationBootstrap {
     }
   }
 
+  async emitPingJobs(projectId: number, pageId: number, profiles: Profile[], envs: Environment[]) {
+    const payloads: CreateJobEvent<JobType.LabPing>[] = []
+
+    for (const profile of profiles) {
+      for (const env of envs) {
+        const key = `${pageId}-${profile.id}-${env.id}`
+        await this.redis.mset(key, 'pending')
```
Comment (EYHN) [inline packages/platform-server/src/modules/snapshot/service.ts:365]: this.redis.set(key, 'pending', 'EX', /* 7 days */ 3600 * 7 * 24)

File: packages/platform/src/modules/project/settings/settings-common-comp.tsx [hunk 2]
```diff
@@ -121,32 +139,43 @@ export const ButtonOperators = <T extends any>(props: ButtonProps<T>) => {
...
-          styles={hideDeleteButton ? undefined : { root: { marginRight: '12px' } }}
-          onClick={onClickRestoreButton}
-        >
-          Restore
-        </DefaultButton>
+        <OperationItemWrap color={SharedColors.cyanBlue10} onClick={onClickRestoreButton}>
+          <div>
+            <RollbackOutlined />
+          </div>
+          <span>Restore</span>
```
Comment (EYHN) [inline packages/platform/src/modules/project/settings/settings-common-comp.tsx:169]: use `enable` maybe more clear

**Review by EYHN — CHANGES_REQUESTED** [review]

**Comment by forehalo** [comment]
reorganize the commit messages @lizimeow

File: packages/platform-server/src/utils/get-lh-run-data.ts [hunk 3]
```diff
@@ -56,3 +56,17 @@ export function getLighthouseRunData(
...
 }
+
+export function getLabPingData(page: Page, profile: Profile, env: Environment): PingJobPayload {
+  const throttle = CONNECTIONS.find((item) => item.id === profile.bandWidth)
+  const deviceId = profile.device
+
+  return {
+    key: `${page.id}-${profile.id}-${env.id}`,
+    url: page.url,
+    throttle: throttle ?? {},
```
Comment (forehalo) [inline packages/platform-server/src/utils/get-lh-run-data.ts:67]: why do we need throttle here?

**Review by EYHN — APPROVED** [review]

**Pull request merged**